### PR TITLE
docs: add dynamic badge

### PR DESCRIPTION
**Pull request by bmorelli25**
## What does this pull request do?

1. This PR adds the `Dynamic` badge to supported configurations:

- [x] `capture_body`
- [x] `TransactionSampleRate`
- [x] `TransactionMaxSpans`

2. It adds a commented out `Dynamic` badge to pending (https://github.com/elastic/apm-agent-dotnet/issues/794) configurations:

* `capture_headers`
* `log_level`
* `span_frames_min_duration`
* `stack_trace_limit`

3. This PR adds a section on `Dynamic configuration`.

**Open questions:**

* `CaptureBodyContentTypes` shows as reloadable, but it isn't supported in the APM app. See: https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/common/agent_configuration/setting_definitions/general_settings.ts. How should we mark this config?
* What do you think of this approach for documenting this feature?

## Documentation preview

http://apm-agent-dotnet_813.docs-preview.app.elstc.co/diff

## Related issues

For https://github.com/elastic/apm/issues/232
For https://github.com/elastic/observability-dev/issues/639

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -143,14 +154,14 @@ NOTE: The service name must conform to this regular expression: `^[a-zA-Z0-9 _-]
...
 |============
-| Environment variable name      | IConfiguration or Web.config key 
+| Environment variable name      | IConfiguration or Web.config key
 | `ELASTIC_APM_SERVICE_NAME` | `ElasticApm:ServiceName`
 |============
 
 [options="header"]
 |============
 | Default                          | Type
-| Name of the entry assembly       | String
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:153]: Sorry about all of these. Just updated my formatting rules. I can remove them from this PR if you'd like.

Comment (gregkalapos) [reply docs/configuration.asciidoc:153]: No worries, all valid changes - feel free to keep those.

**Comment by gregkalapos** [comment]
@v1v just a nit here following up on #793 - in this "doc-only PR" the whole repo gets built still because there is a new `.svg` file. I feel we'll survive this :)  just saying in case you want to reiterate on that.

**Comment by bmorelli25** [comment]
Awesome! If it's alright, I'll wait until tomorrow to merge this. I'm using this PR and a similar one in the Python repository for feedback on this documentation approach before applying it to all Agents.

> Will we still offer a text field in Kibana to put in any config name for central config?

No, only the seven configurations listed in the description will be configurable. Sounds good. I'll leave it unmarked for now.

**Comment by gregkalapos** [comment]
I should have not talked about the CI on this PR... 

The last build reached time-out (1 hour - it was already almost done) 😱  - I have never seen that (currently it takes around 55mins usually - 25 from that is just installing tools). Re-triggered.

**Comment by v1v** [comment]
> @v1v just a nit here following up on #793 - in this "doc-only PR" the whole repo gets built still because there is a new .svg file. I feel we'll survive this :) just saying in case you want to reiterate on that.

https://github.com/elastic/apm-agent-dotnet/pull/817

should help

**Pull request merged**